src/DatePicker/DateDisplay.js

```diff
@@ -128,12 +129,14 @@ class DateDisplay extends Component {
     const {prepareStyles} = this.context.muiTheme;
     const styles = getStyles(this.props, this.context, this.state);
     const year = selectedDate.getFullYear();
+    const regex = /[.,]/;
+    const delimiter = this.props.locale === 'fr' ? '.' : ',';
```

Comment:
I know and I agree. This is what I was thinking when I was considering locales. However I don't think this would do any harm as most of the locales will use either a comma or a dot. Only crazy thing would be if there is a hypen or slash. 

I can do one thing and it's a crude way/ hack. Leave it as it was append &nbsp; after weekday so that it pushes the day and month to the bottom. What say?